### PR TITLE
azure-e2e: Install AKS in parallel to podvm image build

### PR DESCRIPTION
**Pull request by surajssd**
This will reduce the time to run the e2e tests by at least five minutes.

### CODE REVIEW

**Comment by surajssd** [comment]
Code repo to test this PR: https://github.com/surajssd/caa-test2/pull/1 and test runs here: https://github.com/surajssd/caa-test2/actions/workflows/azure-e2e-test.yml

File: .github/workflows/azure-e2e-test.yml [hunk 1]
```diff
@@ -116,42 +119,76 @@ jobs:
...
         ssh-keygen -t rsa -b 4096 -f install/overlays/azure/id_rsa -N "" -C dev@coco.io
 
+    - name: Save the configuration created here
+      uses: actions/cache/save@v3
+      with:
+        path: |
+          install/overlays/azure
+          ${{ env.TEST_PROVISION_FILE }}
+        key: caa-config-${{ github.run_id }}-${{ github.run_attempt }}
+
```
Comment (mkulke) [inline .github/workflows/azure-e2e-test.yml:137]: It's probably more idiomatic to use [artifacts][(https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#passing-data-between-jobs-in-a-workflow)) for this:

```suggestion
    - name: Save the configuration created here
      uses: actions/upload-artifact@v3
      with:
        path: |
          install/overlays/azure
          ${{ env.TEST_PROVISION_FILE }}
        name: e2e-configuration
```

File: .github/workflows/azure-e2e-test.yml [hunk 2]
```diff
@@ -116,42 +119,76 @@ jobs:
...
+      with:
+        go-version: '1.20'
+
+    - name: Restore the configuration created before
+      uses: actions/cache/restore@v3
+      with:
+        path: |
+          install/overlays/azure
+          ${{ env.TEST_PROVISION_FILE }}
+        key: caa-config-${{ github.run_id }}-${{ github.run_attempt }}
```
Comment (mkulke) [inline .github/workflows/azure-e2e-test.yml:174]:
```suggestion
    - name: Restore the configuration created before
        uses: actions/download-artifact@v3
        with:
          name: e2e-configuration
```

File: .github/workflows/azure-e2e-test.yml [hunk 3]
```diff
@@ -165,13 +202,34 @@ jobs:
...
```
Comment (mkulke) [inline .github/workflows/azure-e2e-test.yml:216]: uses: actions/download-artifact@v3
        with:
          name: homework
```suggestion
    - name: Restore the configuration created before
      uses: actions/download-artifact@v3
      with:
        name: e2e-configuration
```

**Comment by surajssd** [comment]
Here is a successful run from the recent changes: https://github.com/surajssd/caa-test2/actions/runs/6424587421/job/17446149600

**Comment by surajssd** [comment]
@mkulke @kartikjoshi21  any other thoughts on this?

**Review by mkulke — APPROVED** [review]

**Review by kartikjoshi21 — APPROVED** [review]

**Pull request merged**